otelx/middleware.go

```diff
@@ -0,0 +1,66 @@
+package otelx
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/julienschmidt/httprouter"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const tracingComponent = "github.com/ory/x/otelx"
+
+func isHealthFilter(r *http.Request) bool {
+	path := r.URL.Path
+	if strings.HasPrefix(path, "/health/") {
+		return false
+	}
+	return true
+}
+
+func filterOpts() []otelhttp.Option {
+	filters := []otelhttp.Filter{
+		isHealthFilter,
+	}
+	opts := []otelhttp.Option{}
+	for _, f := range filters {
+		opts = append(opts, otelhttp.WithFilter(f))
+	}
+	return opts
+}
+
+func NewHandler(handler http.Handler, operation string) http.Handler {
+	return otelhttp.NewHandler(handler, operation, filterOpts()...)
+}
+
+// Middleware to satisfy httprouter.Handle. The URL path is used as
+// the span name.
+func WrapHTTPRouter(next httprouter.Handle) httprouter.Handle {
```

Comment:
This works of course, but wouldn't it be nicer to implement the [negroni middleware](https://pkg.go.dev/github.com/urfave/negroni#Handler) interface so that it works independed of `httprouter`?

Reply:
I just realized that otelx.WrapHTTPRouter isn't really required since the actual http.Handler above is already being wrapped. Our tracing works just fine without this.